crates/protogen/proto/metastore/catalog.proto

```diff
@@ -49,8 +49,13 @@ message CatalogState {
 
   // Metadata for the deployment.
   DeploymentMetadata deployment = 3;
-
   // next: 4
+
+  // The version of the actual catalog implementation.
+  // This is incremented when there are incompatible changes to the physical
+  // catalog layout.
+  // Purposely set to a high number to avoid accidental collisions
+  optional uint32 catalog_version = 999;
```

Comment:
I'm not particularly worried about collisions: any collison that gets past the compiler and code review is buggy in more ways than one, and setting this here only protects against well behaved accidents, and collisions here are definitionally not well behaved. 

Given var-length encoding, and other peculiarities, https://protobuf.dev/programming-guides/proto3/#assigning, I'd be inclined to make this 2047 (the largest two-byte encoded integer)